2023-02-08_running-ruby-locally.md

```diff
@@ -0,0 +1,82 @@
+# Running Ruby On Your Own Machine
+
+
+## Why?
+Now that we've been using ruby in gitpod and treehouse it's time to start running ruby on your own machine.
+1. more typical day to day developer experience
+2. helpful for learning
+3. fun to start hacking your own scripts
+
+
+## Terminal Commands
+macOS is in the unix family so we have these useful commands
+* `ls` - list directory contents
+* `pwd` - print working directory
+* `mkdir` - make directory
+* `cd` - change directory
+* `touch` - updates last open time of file
+* `open` - open files and directories
+* Use `man` to learn more about each command
+
+
+## Interactive Ruby (irb)
+open terminal
+* command + space (open spotlight search)
+* search 'terminal', press `enter`
+* type in `irb`
+* play around with ruby
+* type `exit` to exit irb shell
+
+
+## Setup vscode
+code editor
+works great with ruby, rails, and pretty much any language
+
+* [install vscode](https://code.visualstudio.com/)
+* open vscode
+* open 'command palette' (command + shift + p)
+* add `code .` to your bash profile
+  * this way we can open vscode from the terminal
+  * `.` is shorthand for current directory
+
+
+## Create Workspace Folder
+* open terminal (command + space, search terminal)
+* create workspace for all your different projects `mkdir Workspace`
+* navigate to workspace `cd Workspace`
```

Comment:
Yes, that's why I do it. Tradeoff between being consistent and not adding capitals and spaces to directory names. I have them add a `scratch/` directory using lower case. Maybe I can just think out loud and give context.

Reply:
I can definitely make a point on ruby version. Show how to check `ruby -v` Probably too much to demo adding different versions using rbenv, rvm etc.

Reply:
I added a demo `hello.rb` section which takes name input and replies `"hello #{name}"`. Give them option to make it their own. Can add option to move over treehouse programs.

Reply:
> Can add option to move over treehouse programs.

Maybe relevant:

https://piazza.com/class/ldj532ul5a0621/post/ldxaz2gw9ve7o7

They could download them from Treehouse and run them directly

Reply:
> Probably too much to demo adding different versions using rbenv, rvm etc.

Yes, agreed that we don't need to mess with version managers right now. It will come later.